src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelSelectorField.tsx

```diff
@@ -46,13 +51,105 @@ class DataPrivacyRulesPanelSelectorField extends React.Component<Props, State> {
     this.loadFieldValues(this.props.value, true);
   }
 
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.selectorSuggestions != this.props.selectorSuggestions) {
+      this.loadFieldValues(this.props.value, false);
+    }
+  }
+
   componentWillUnmount() {
     document.removeEventListener('mousedown', this.handleClickOutside, false);
   }
 
   selectorField = React.createRef<HTMLDivElement>();
   suggestionList = React.createRef<HTMLUListElement>();
 
+  getAllSuggestions() {
+    return [...this.getValueSuggestions(), ...unaryOperatorSuggestions, ...binaryOperatorSuggestions];
+  }
+
+  getValueSuggestions() {
+    return this.props.selectorSuggestions;
+  }
+
+  getFilteredSuggestions = (value: string, type: SuggestionType) => {
+    let valuesToBeFiltered: Array<Suggestion> = [];
+
+    switch (type) {
+      case 'binary': {
+        valuesToBeFiltered = binaryOperatorSuggestions;
+        break;
+      }
+      case 'value': {
+        valuesToBeFiltered = this.getValueSuggestions();
+        break;
+      }
+      case 'unary': {
+        valuesToBeFiltered = unaryOperatorSuggestions;
+        break;
+      }
+      default: {
+        valuesToBeFiltered = [...this.getValueSuggestions(), ...unaryOperatorSuggestions];
+      }
+    }
+
+    const filteredSuggestions = valuesToBeFiltered.filter(
+      s => s.value.toLowerCase().indexOf(value.toLowerCase()) > -1
+    );
+
+    return {
+      filteredSuggestions,
+      showSuggestions: !(
+        filteredSuggestions.length === 1 && filteredSuggestions[0].value === value
+      ),
+    };
+  };
+
+  getNewSuggestions = (fieldValues: Array<Suggestion | Array<Suggestion>>) => {
```

Comment:
I'd prefer to move some functions to another file, because this is already too large

Reply:
I had to move those functions back because they all access `this` now. I'm not sure how to split them up?